Architecture mismatch causes Lambda function failure when deploying from x86 machine

### Issue Description

When deploying the CDK project from an x86 machine, the Lambda function fails to execute due to an architecture mismatch. The project appears to be designed for deployment from an ARM-based machine (likely an Apple Silicon MacBook), but when deployed from an x86 machine, it creates incompatible binaries for the ARM-based Lambda environment.

### Error Message

The Lambda function throws the following error:

```
[ERROR] ImportError: /var/task/pymupdf/_extra.so: cannot open shared object file: No such file or directory
Traceback (most recent call last):
File "/var/task/myapp.py", line 137, in lambda_handler
import fitz
File "/var/task/fitz/init.py", line 2, in <module>
from pymupdf import *
File "/var/task/pymupdf/init.py", line 30, in <module>
from . import extra
File "/var/task/pymupdf/extra.py", line 10, in <module>
from . import _extra
```

This error occurs because the `_extra.so` file, which is a compiled binary, is built for x86 architecture during the deployment process on an x86 machine. However, the Lambda function runs in an ARM environment, causing the incompatibility.

### Root Cause

The CDK deployment process builds certain dependencies, including the `pymupdf` library, during the deployment. When this occurs on an x86 machine, it creates x86-compatible binaries. However, the Lambda function is configured to use ARM architecture, leading to the inability to load the x86-compiled `_extra.so` file.

### Proposed Solutions

1. **Update Deployment Instructions**: 
   Add a clear note in the deployment instructions specifying that the project should be deployed from an ARM-based machine (e.g., Apple Silicon MacBook) to ensure compatibility with the Lambda function's ARM environment.

2. **Implement Architecture Check**:
   Modify the CDK deployment script to detect the architecture of the deployment machine. If it's x86, either:
   a) Warn the user and suggest using an ARM-based machine for deployment, or
   b) Automatically adjust the Lambda function's architecture to match the deployment machine (i.e., set it to x86_64 if deploying from an x86 machine).

### Steps to Reproduce

1. Clone the repository on an x86 machine.
2. Follow the current deployment instructions using CDK.
3. Observe that the Lambda function fails with the above error when triggered.

### Environment Details

- Deployment Machine: x86 architecture (e.g., Intel-based MacBook or PC)
- Target Lambda Environment: ARM64
- AWS CDK Version: 2.166.0
- Node.js Version: v18.20.2

--- a/app.py
+++ b/app.py
@@ -17,12 +17,12 @@ from aws_cdk import (
     aws_secretsmanager as secretsmanager
 )
 from constructs import Construct
-
+import platform
 
 class PDFAccessibility(Stack):
     def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
         super().__init__(scope, construct_id, **kwargs)
-
+        
         # S3 Bucket
         bucket = s3.Bucket(self, "pdfaccessibilitybucket1", encryption=s3.BucketEncryption.S3_MANAGED, enforce_ssl=True)
     
@@ -230,6 +230,13 @@ class PDFAccessibility(Stack):
         # //
 
         # Define the Add Title Lambda function
+        host_machine = platform.machine().lower()
+        print("Architecture of Machine:",host_machine)
+        if "arm" in host_machine:
+            lambda_arch = lambda_.Architecture.ARM_64
+        else:
+            lambda_arch = lambda_.Architecture.X86_64
+
         add_title_lambda = lambda_.Function(
             self, 'AddTitleLambda',
             runtime=lambda_.Runtime.PYTHON_3_12,
@@ -237,7 +244,8 @@ class PDFAccessibility(Stack):
             code=lambda_.Code.from_docker_build('lambda/add_title'),
             timeout=Duration.seconds(900),
             memory_size=1024,
-            architecture=lambda_.Architecture.ARM_64
+            # architecture=lambda_.Architecture.ARM_64
+            architecture=lambda_arch,
         )
 
         # Grant the Lambda function read/write permissions to the S3 bucket

